netconf/transport_junos.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2018, Juniper Networks, Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+/*
+Transport JunOS provides the ability to communicate with JunOS via local shell
+NETCONF interface (xml-mode netconf need-trailer).
+*/
+package netconf
+
+import (
+	"os/exec"
+)
+
+// TransportJunOS maintains the information necessary to communicate with JunOS
+// via local shell.
+type TransportJunOS struct {
+	transportBasicIO
+	cmd *exec.Cmd
+}
+
+// Close closes an existing local NETCONF session.
+func (t *TransportJunOS) Close() error {
```

Comment:
You need to  kill the process here as well otherwise it will leak

Reply:
The process is closed by `t.ReadWriteCloser.Close()`. I checked on the Junos device that child process was removed successfully. Trying to call `t.cmd.Process.Kill()` returns an error with `operation not permitted`.